jsonschema-core.md

```diff
@@ -2071,70 +2071,44 @@ Omitting this keyword has the same assertion behavior as an empty schema.
+vernacular across differing formats. Output specifications which use this
+information MUST use this terminology to describe it. Conversely, output
+specifications which use these terms MUST maintain their meaning.
 
-### Output Formats
+### Evaluation path
 
-This specification defines three output formats. See the "Output Structure"
-section for the requirements of each format.
-
-- *Flag*: A boolean which simply indicates the overall validation result with no
-  further details.
-- *List*: Provides validation information in a flat list structure.
-- *Hierarchical*: Provides validation information in a hierarchical structure that
-follows the evaluation paths generated while processing the schema.
-
-An implementation MUST provide the "flag" format and SHOULD provide at least one
-of the "list" or "hierarchical" formats. Implementations SHOULD specify in their
-documentation which formats they support.
-
-### Minimum Information
-
-Beyond the simplistic "flag" output, additional information is useful to aid in
-debugging a schema or instance. Each sub-result SHOULD contain the information
-contained within this section at a minimum.
-
-A single object that contains all of these components is considered an output
-unit.
-
-Implementations MAY elect to provide additional information.
-
-#### Evaluation path
-
-The evaluation path to the schema object that produced the output unit. The
-value MUST be expressed as a JSON Pointer, and it MUST include any by-reference
-applicators such as `$ref` or `$dynamicRef`.[^13]
-
-[^13]: The schema may not actually have a value at the location indicated by
-this pointer. It is provided as an indication of the traversal path only.
+The evaluation path is the set of keys, starting from the schema root, through
+which evaluation passes to reach the schema object that produced a specific
+result. The value MUST be expressed as a JSON Pointer, and it MUST include any
+by-reference applicators such as `$ref` or `$dynamicRef`.
 
 ```
 /properties/width/$ref/allOf/1
 ```
 
-Note that this pointer may not be resolvable by the normal JSON Pointer process
-due to the inclusion of these by-reference applicator keywords.
+Note that this pointer may not be resolvable on the root schema by the normal
+JSON Pointer process. It is intended as an indication of the traversal path
+only.
 
-The JSON key for this information is "evaluationPath".
+When represented in JSON, the key for this information MUST be "evaluationPath".
 
-#### Schema Location
+### Schema Location
 
-The absolute, dereferenced location of the schema object that produced the
-output unit. The value MUST be expressed using the canonical IRI of the relevant
-schema resource plus a JSON Pointer fragment that indicates the schema object
-that produced the output. It MUST NOT include by-reference applicators such as
-`$ref` or `$dynamicRef`.[^14]
+The schema location is the absolute, dereferenced location of the schema object
```

Comment:
It occurred to me that I don't think the word "dereferenced" makes sense here. "Dereferenced" usually means getting a representation from a URI, but this value is a URI, not schema. Maybe the word "normalized" was intended?

Reply:
Given our recent conversations on URIs and the word "location," I'm open to renaming this.  However, I'd prefer we discuss that elsewhere and leave this for now.

I think "dereferenced" is correct because it indicates that any references have been resolved and this is the most direct URI that leads to the (sub)schema (the canonical URI).

Reply:
I think strictly, this is the correct use of "dereferenced", as it relates to URIs as opposed to a `$ref`. People use it in error in relation to a `$ref` when they mean "transclusion". What they really mean is "de-referencing" as in taking out references".
This is my understanding when reading RFC 3986. Please correct me if I'm wrong.

Reply:
> This is my understanding when reading RFC 3986. Please correct me if I'm wrong or you have a different understanding.

From RFC 3986 ...

> Given a URI, a system may attempt to perform a variety of operations on the resource, as might be characterized by words such as "access", "update", "replace", or "find attributes".  Such operations are defined by the protocols that make use of URIs, not by this specification.  However, we do use a few general terms for describing common operations on URIs.  URI "resolution" is the process of determining an access mechanism and the appropriate parameters necessary to dereference a URI; this resolution may require several iterations.  **_To use that access mechanism to perform an action on the URI's resource is to "dereference" the URI._**

That's how I understand the term and I don't see how that applies in this situation.

> I think "dereferenced" is correct because it indicates that any references have been resolved and this is the most direct URI that leads to the (sub)schema (the canonical URI).

Sorry, I'm not following what you mean. What would this value look like if it wasn't "dereferenced"?

Reply:
> What would this value look like if it wasn't "dereferenced"?

I'm saying that it shouldn't look like the evaluation path, which explicitly preserves any reference keywords that were followed.

It should be the schema's normalized canonical IRI _plus_ a JSON Pointer fragment (which might be empty) to the specific subschema location.

So it's both dereferenced _and_ normalized, but not "absolute" in the RFC 3986 sense of an absolute URI.  I'm happy to use both terms...

Reply:
Got it. Now I understand what it's trying to say. It just means that it should be the canonical URI for the schema. Since the intended meaning of both "absolute" and "dereferenced" is unclear and saying it's the canonical URI says everything it needs to say, I suggest dropping both of those words.